### PR TITLE
test: remove v1beta1 CRD/RBAC e2e tests

### PR DESCRIPTION
**Pull request by sdowell**
These tests all require k8s versions less than 1.22, which is no longer included in our testing matrix. As these tests are no longer exercised, they can be removed.

### CODE REVIEW

File: e2e/testcases/custom_resource_definitions_test.go [hunk 1]
```diff
@@ -210,14 +203,6 @@ func TestAddAndRemoveCustomResourceV1(t *testing.T) {
 	addAndRemoveCustomResource(nt, "v1_crds", "anvil-crd.yaml")
 }
 
-func TestAddAndRemoveCustomResourceV1Beta1(t *testing.T) {
-	nt := nomostest.New(t, nomostesting.Reconciliation1)
-	if !nt.SupportV1Beta1CRDAndRBAC() {
-		nt.T.Skip("Kubernetes v1.22 and later do not support the v1beta1 CRD API")
-	}
-	addAndRemoveCustomResource(nt, "v1beta1_crds", "anvil-crd-v1.yaml")
```
Comment (nan-yu) [inline e2e/testcases/custom_resource_definitions_test.go:218]: The `e2e/testdata/customeresources/v1beta1_crds` folder can be deleted as well.

Comment (sdowell) [reply e2e/testcases/custom_resource_definitions_test.go:218]: done

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**